syllabus/core/curriculum_base.py

```diff
@@ -149,14 +152,31 @@ def _sample_distribution(self) -> List[float]:
         """
         raise NotImplementedError
 
-    def _should_use_startup_sampling(self) -> bool:
-        return self.random_start_tasks > 0 and self.completed_tasks < self.random_start_tasks
-
-    def _startup_sample(self) -> List:
-        task_dist = [0.0 / self.num_tasks for _ in range(self.num_tasks)]
-        task_dist[0] = 1.0
-        return task_dist
-
+    def _should_use_startup_sampling(self, flag: str) -> bool:
+        if flag == "finish":
+            return False
+        return True
+    
+    def _startup_sample(self, k: int) -> List:
+        if self.use_startup_behavior:
+            sampled_tasks_per_env = [[] for _ in range(k)]
+            for env_index in range(k):  
+                for behavior, episode, tasks in self.sampled_list[env_index]:
+                    if behavior == "fix":
+                        for _ in range(episode):
+                            end_index = min(self.current_task_index + tasks, len(self.tasks))
+                            sampled_tasks_per_env[env_index].extend(self.tasks[self.current_task_index:end_index])
+                            self.current_task_index = end_index
+
+                    elif behavior == "random":
+                        for _ in range(episode):
+                            indices = np.random.choice(len(self.tasks), size=tasks, replace=False if tasks <= len(self.tasks) else True)
+                            sampled_tasks_per_env[env_index].extend([self.tasks[idx] for idx in indices])
```

Comment:
I think you can pretty much remove the two loops here and try again, it should be much simpler than this. For "random" behavior you just want to sample from a uniform distribution (you can probably steal code from the DomainRandomization curriculum). For "fixed" you want to get the list of tasks (self.tasks) and return the next k elements after current_task_index.